### ISSUE TITLE
feat(prompts) - Ability for user to see `instruction-patterns` from prompts

### ISSUE THREAD

**Issue by jduckles** [stencila/stencila#2435]
### Which component?

System (and eventually user / community) prompts.

### What's your idea?

There is magic involved in prompts being automatically matched based on particular regular expression patterns. The mechanism for this is to parse the user prompt string for matches to the various `instruction-patterns` in each prompt yaml header. These patterns are spread across tens of files, and there is no single source for making these patterns visible. 

At a minimum - raising these *magic* keywords / patterns to a place the user can see in the documentation for system prompts (and not have to parse a regular expression with their brain). 

Further down the track - Being able to mouse over prompts in the VSCode side-bar and seeing a kind of long descriptive docstring of the prompt which includes human-readable versions of `instruction-patterns` 

Stretch goal - It would be nice to wire up some kind of code-lense VSCode integration that highlights the usage of these patterns as being keywords to drive the user to a specific prompt. 

As this system is intended to be extensible with the user able to create and override system prompts. The system for raising this prompt visibility should be aware of user prompts as well.

**Comment by nokome** [comment]
Agreed, but I think the most important thing here is for the user to be able to see, **before** running the command, which prompt has been inferred from their instruction, and to be able to override that if necessary by explicitly selecting a specific prompt. Right now, the prompt is resolves just-in-time, so the user has to actually run the command to see which prompt is/was used, and **then** change it.

I think that as a first step, that is more important than showing the details of **how** the prompt was guessed by showing regexes. This is partly because the **how** may change: we started with regexes, moved to semantic similarity ranking, and went back to regexes because it was simpler to deploy (we had to ship the ONNX runtime for the semantic approach).

Adding a code lens which shows the currently implied prompt (based on users instruction, as they type) and brings up the current "Prompt picker" command is likely to be straightfoward.
